openff/toolkit/topology/molecule.py

```diff
@@ -3944,7 +3944,9 @@ def _add_bond(
             The index of the bond in the molecule
 
         """
-        if isinstance(atom1, int) and isinstance(atom2, int):
+        if isinstance(atom1, (int, np.integer)) and isinstance(
+            atom2, (int, np.integer)
+        ):
```

Comment:
Are you sure the deprecation warning is coming from `np.integer` and not `np.int` (somewhere else)? The latter was deprecated in [v1.20.0](https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations). Anyway, this can be easily moved to the `mmic_openff` pkg.

Reply:
`isinstance(atom1, numbers.Integral)` should be enough, see: https://docs.python.org/3/library/numbers.html https://github.com/numpy/numpy/pull/4547